libs/failsafe-controller/src/main/java/net/jodah/failsafe/FailsafeController.java

```diff
@@ -89,9 +105,14 @@ public String getId() {
    *
    * @param retryPolicy the retry policy to use
    * @throws NullPointerException if {@code retryPolicy} is null
+   * @throws IllegalStateException if any of the <code>retry()</code> methods have been called
+   *     already
    */
-  public SyncFailsafe<R> with(RetryPolicy retryPolicy) {
-    return new SyncBuilder(retryPolicy);
+  public synchronized SyncFailsafe<R> with(RetryPolicy retryPolicy) {
+    Assert.state(sync == null, "with() was already called once");
+    this.sync = new SyncBuilder(new FailsafeConfig<R, FailsafeConfig<R, ?>>());
```

Comment:
ℹ️ Because the SyncBuilder now extends the sync delegater class to inherit the asynchronous listener handling logic, we simply create a failsafe config to be used as the master for all syncs and asyncs that will be created from this one.